constants/allow_events.go

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) 2023 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package constants
+
+// Allowed repo events.
+const (
+	AllowPushBranch = 1 << iota // 00000001 = 1
+	AllowPullOpen               // 00000010 = 2
+	AllowPullEdit               // 00000100 = 4
+	AllowPullSync               // ...
+	AllowPushTag
+	AllowDeployCreate
+	AllowCommentCreate
+	AllowCommentEdit
```

Comment:
A note on this: we can't shuffle these at any point if this were to be implemented. So, if a `pull_request` event with the action `review_requested` is something we support later, then it must go at the bottom.

Reply:
seems like a good thing to add to comments in-line

Reply:
Can we introduce some kind of version bit (byte?) so that the entire thing can be invalidated in the future? For now, just include the bit. Then, the migration logic can be introduced when we bump that version, if we ever do so.

Reply:
Since we're using Intl64, and so far this only uses 8 bits, So, the version info could actually be at the end of the 64 bits instead of the beginning. Then, we don't need to reserve any space.

Reply:
My hesitation with including a version is that it would require some sort of DB migration script that updates all existing repos every time we add a new event we support. Part of me wants to just go in and enumerate all possible webhook actions that Vela could theoretically subscribe to and just mark them as `not implemented` if they aren't. That way future enhancements would build off a clean order and it wouldn't change unless GitHub itself adds a new event / action worth implementing.

Reply:
Yes. Let's leave room for all current GitHub events, but leave a `_` for all of the unimplemented ones with a comment saying what that slot is reserved for. Maybe even leave a bit of a gap in each category to add more later.

Reply:
Added space for future implementations for `pull_request` actions that seem potentially relevant to CI

Reply:
There seems to be a lot of other possible webhook events:
https://docs.github.com/en/webhooks/webhook-events-and-payloads

How did you pick which of the `pull_request` actions deserved an entry here?